File: Simulation/Bronze/P2_2018-The_Bucket_List/blist.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
	freopen("blist.in", "r", stdin);
	// Read in first line: n = # of buckets
	int n;
	cin >> n;
	// For the next n lines, read in the start, end, and buckets needed
	int inUse[1001] = {0};
	for (int i = 0; i < n; i++) {
		int start, end, bucketsNeeded;
		cin >> start >> end >> bucketsNeeded;
		for (int j = start; j <= end; j++)
			inUse[j] += bucketsNeeded;
	}

	freopen("blist.out", "w", stdout);
	cout << *max_element(inUse, inUse + 1001) << endl;
	return 0;
}

File: Simulation/Bronze/P2_2018-The_Bucket_List/blist.py
with open("blist.in", 'r') as fin:
    # Read in first line: n = # of buckets
    n = int(fin.readline())
    # For the next n lines, read in the start, end, and buckets needed
    in_use = [0] * 1001
    for _ in range(n):
        start, end, buckets_needed = map(int, fin.readline().split())
        for j in range(start, end + 1):
            in_use[j] += buckets_needed

print(max(in_use), file=open("blist.out", 'w'))
